src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -668,26 +675,14 @@ impl<'a, 'tcx> LinkCollector<'a, 'tcx> {
                     };
 
                     if extra_fragment.is_some() {
-                        Err(ErrorKind::AnchorFailure(AnchorFailure::RustdocAnchorConflict(ty_res)))
+                        Err(ErrorKind::AnchorFailure(AnchorFailure::RustdocAnchorConflict(root_res)))
                     } else {
                         let res = Res::Def(item.kind.as_def_kind(), item.def_id);
-                        Ok((res, Some(format!("{}.{}", kind, item_str))))
+                        Ok((res, Some(format!("{}.{}", kind, item_name))))
```

Comment:
I just want to point out that according to the Symbol docs, this is a "slowish operation because it requires locking the symbol interner", because `Symbol::fmt` is implemented using `as_str()`. No big deal, though.